### PR TITLE
Bocaj - Third Annual Refactor (Redux)

### PR DESCRIPTION
**Pull request by JacobJerrell**
## Description

Original PR: #10146 
The previous PR had somehow gotten changes from a submodule mixed in with it. To ultimately resolve the issue I had to reset this branch to upstream/master and then reapply my changes. Much to my surprise, this action (and possibly the force push specifically) appears to have closed/removed the original PR

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by JacobJerrell** [comment]
@drashna please see above for an explanation. Sorry for the inconvenience!

**Review by drashna — APPROVED** [review]

File: layouts/community/ergodox/bocaj/config.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #pragma once
-
 #include QMK_KEYBOARD_CONFIG_H
```
Comment (drashna) [inline layouts/community/ergodox/bocaj/config.h:2]: This isn't needed, nor even actually used/generated and should be removed. 
```suggestion
```

File: layouts/community/ergodox/bocaj/config.h [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 
 #define LEADER_TIMEOUT 250
 #define LEADER_PER_KEY_TIMING
+#undef TAP_DANCE_ENABLE
```
Comment (drashna) [inline layouts/community/ergodox/bocaj/config.h:6]: The proper way for this is to have `TAP_DANCE_ENABLE = no` in your rules.mk.
```suggestion
```

File: users/bocaj/bocaj.h [hunk 3]
```diff
@@ -16,70 +17,55 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #pragma once
-#include "quantum.h"
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline users/bocaj/bocaj.h:20]: If you have this here, and this file included in the keymap.c files... then you don't need this include in those files.

Comment (JacobJerrell) [reply users/bocaj/bocaj.h:20]: Updates have been applied to the keymap.c files

File: layouts/community/ortho_4x12/bocaj/rules.mk [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        RGBLIGHT_STARTUP_ANIMATION  = yes
+    endif
+else
+    CONSOLE_ENABLE    			= yes
+    COMMAND_ENABLE    			= yes
+    RGBLIGHT_ENABLE             = yes
+    RGBLIGHT_STARTUP_ANIMATION  = yes
+    RGB_MATRIX_ENABLE           = no
+    AUDIO_ENABLE       			= yes
+    EEPROM_DRIVER 				= i2c
```
Comment (drashna) [inline layouts/community/ortho_4x12/bocaj/rules.mk:24]: Just a heads up, if you're using the rev6 with this setting and don't actually have an eeprom chip hooked up...

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]
All code limited to userspace or user layouts, assuming it's all working as intended.

**Pull request merged**